app/controllers/application_controller.rb

```diff
@@ -100,13 +100,22 @@ def service_provider_request
   end
 
   def after_sign_in_path_for(_user)
-    user_session[:stored_location] || sp_session[:request_url] || signed_in_url
+    user_session[:stored_location] || sp_session_url || signed_in_url
   end
 
   def signed_in_url
     user_fully_authenticated? ? account_or_verify_profile_url : user_two_factor_authentication_url
   end
 
+  def sp_session_url
```

Comment:
How would you feel about naming this to something like `after_sign_in_sp_destination_url` since it can return either the SP session URL or the sign up completed url?